compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -864,6 +867,7 @@ public void visit(BLangMatchStatement matchStatement) {
                     containsLastPatternInStatement || (matchClause.matchGuard == null && this.hasLastPatternInClause);
         }
         this.statementReturns = allClausesReturns && containsLastPatternInStatement;
+        resetLastPatternInClause();
```

Comment:
Don't think we need a method here. If we want a method it is good if we can make [line 850](https://github.com/ballerina-platform/ballerina-lang/pull/27343/files#diff-6b4a8cb1f7bd56d5a95a80500c4febcba531cdde5af59b66a477acd35e8485acR850) to a method call to make things consistent.

Reply:
IMO, it's OK to have a method, but yeah I agree that we should then change other usage too.

Reply:
This is no longer used.